src/providers/LocalizationProvider.tsx

```diff
@@ -48,54 +48,62 @@ export default function LocalizationProvider({
+    }
   }, [dispatch, selectedLocale]);
 
   useEffect(() => {
-    if (countriesResponse && countriesResponse.status === 'success' && countriesResponse.data) {
+    if (selectedLocale && countriesResponse && countriesResponse.status === 'success' && countriesResponse.data) {
       const countriesCopy = [...countriesResponse.data];
       setCountries(countriesCopy.sort((a, b) => a.name.localeCompare(b.name, selectedLocale)));
     }
   }, [selectedLocale, countriesResponse]);
 
   useEffect(() => {
-    if (timeZoneResponse && timeZoneResponse.status === 'success' && timeZoneResponse.data) {
+    if (selectedLocale && timeZoneResponse && timeZoneResponse.status === 'success' && timeZoneResponse.data) {
       const timezonesCopy = [...timeZoneResponse.data];
       setTimeZones(timezonesCopy.sort((a, b) => a.longName.localeCompare(b.longName, selectedLocale)));
     }
   }, [selectedLocale, timeZoneResponse]);
 
   useEffect(() => {
-    const fetchStrings = async () => {
-      const language = selectedLocale.replace(/[-_].*/, ''); // 'en-US' => 'en'
-      const localeDetails =
-        supportedLocales.find((details) => details.id === selectedLocale) ||
-        supportedLocales.find((details) => details.id === language) ||
-        supportedLocales[0];
-
-      const localeMap: ILocalizedStringsMap = {};
-      localeMap[selectedLocale] = (await localeDetails.loadModule()).strings;
-      strings.setContent(localeMap);
-      strings.setLanguage(selectedLocale);
-
-      setActiveLocale(selectedLocale);
-    };
-
-    void fetchStrings();
+    if (selectedLocale) {
+      const fetchStrings = async () => {
+        const language = selectedLocale.replace(/[-_].*/, ''); // 'en-US' => 'en'
+        const localeDetails =
+          supportedLocales.find((details) => details.id === selectedLocale) ||
+          supportedLocales.find((details) => details.id === language) ||
+          supportedLocales[0];
+
+        const localeMap: ILocalizedStringsMap = {};
+        localeMap[selectedLocale] = (await localeDetails.loadModule()).strings;
+        strings.setContent(localeMap);
+        strings.setLanguage(selectedLocale);
+
+        setActiveLocale(selectedLocale);
+      };
+
+      void fetchStrings();
+    }
   }, [selectedLocale, setActiveLocale, supportedLocales]);
 
-  const context: ProvidedLocalizationData = {
-    activeLocale,
-    countries,
-    bootstrapped: !!activeLocale,
-    selectedLocale,
-    setSelectedLocale,
-    supportedTimeZones: timeZones,
-  };
+  if (selectedLocale) {
+    const context: ProvidedLocalizationData = {
+      activeLocale,
+      countries,
+      bootstrapped: !!activeLocale,
+      selectedLocale,
+      setSelectedLocale,
+      supportedTimeZones: timeZones,
+    };
 
-  return <LocalizationContext.Provider value={context}>{children}</LocalizationContext.Provider>;
+    return <LocalizationContext.Provider value={context}>{children}</LocalizationContext.Provider>;
+  } else {
+    return null;
```

Comment:
I have another version of this change where I updated `ProvidedLocalizationData` such that the selected locale could be null, and then added `if (selectedLocale)` blocks in various places in the code. But this seemed both simpler and safer: we can't really show any UI at all until we know the locale, so there's no point even rendering the child objects before we could display any text.

If there are bad side effects of this approach, though, I can switch to the other one.

Reply:
We default to English if the user doesn't have a locale. In practice, the locale gets set right away when the user logs in, but it needs to be nullable in the user details because when a new user is invited to an organization, we create a user record for them but we don't know their locale preference yet.